packages/grid/_modules_/grid/hooks/features/columns/gridColumnsUtils.ts

```diff
@@ -100,6 +100,17 @@ export const getGridColDef = (
   if (!type) {
     return columnTypes[DEFAULT_GRID_COL_TYPE_KEY];
   }
+
+  if (!columnTypes[type]) {
```

Comment:
If we only console log, we can reduce the bundle size in production

```suggestion
  if (process.env.NODE_ENV !== 'production') {
    if (!columnTypes[type]) {
```